test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore3x/NonEdm/NonEdmTests.cs

```diff
@@ -0,0 +1,159 @@
+//-----------------------------------------------------------------------------
+// <copyright file="NonEdmTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.NonEdm
+{
+    public class NonEdmTests : WebHostTestBase
+    {
+        public NonEdmTests(WebHostTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            configuration.AddControllers(typeof(CustomersController));
+            configuration.Select().Filter().OrderBy().Expand().Count().MaxTop(null);
+
+            // Force use of System.Text.Json output formatter
```

Comment:
@KenitoInc Because in the test fixture, `AddNewtonsoftJson` method is called and that causes the Newtonsoft.Json formatters to have highest precedence: https://github.com/OData/WebApi/blob/2c211035d8e25e193d1c5a3b58999581cfa05a4d/test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Common/Execution/WebHostTestFixture.cs#L239

By deliberately inserting the System.Text.Json output formatter at index 0 of the formatters collection, we ensure that the formatters I added are used for the tests in this test file particularly.